Games/Games/Program.cs

```diff
@@ -0,0 +1,138 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Games.Players;
+
+namespace Games
+{
+    public class Program
+    {
+        readonly static object lockerNotepad = new NotepadPlayer();
+        readonly static object lockerUsual = new UsualPlayer();
+        readonly static object lockerUber = new UberPlayer();
+        readonly static object lockerCheater = new CheaterPlayer();
+        readonly static object lockerUB = new UberCheaterPlayer();
+
+        readonly Mutex mutex = new Mutex();
+
+        public static int basketWeight = new Random().Next(40, 140);
+
+        public static void Main(string[] args)
+        {
+            Console.WriteLine($"Basket weight: {basketWeight}\n\n");
```

Comment:
You did it too complicated.
Please look at the attached example and revise your solution 
`public class Program
    {
        readonly static object locker = new object();
        public static void Main(string[] args)
        {
            var numberOfAttempts = 100;
            var basketWeight = new Random().Next(40, 140);
            var playersList = new List<BasePlayer>()
            {
                new GamePlayer(),
                new UberPlayer(),
                new UsualPlayer(),
                new UberCheaterPlayer(),
                new CheaterPlayer()
            };

            Console.WriteLine("Basket weight: " + basketWeight);
            Console.WriteLine("-------------------");
            bool isGuessed = false;
            int counter = 0;
            while (!isGuessed && counter < numberOfAttempts)
            {
                foreach (var player in playersList)
                {
                    var guessing = new Task<int>(() => player.GuessNumber());
                    
                        guessing.Start();
                        var guessedNumber = guessing.Result;
                        player.PrintCurrentResult(guessedNumber);
                        lock (locker)
                        {
                            if (guessedNumber == basketWeight)
                            {
                                Console.WriteLine(player.ToString());
                                isGuessed = true;
                                return;
                            }
                        }
                }

                counter += playersList.Count;
                Console.WriteLine("-------------------");
            }

            if (!isGuessed)
            {
                Console.WriteLine("Nobody guessed");
            }

        }`